app/models/destination.rb

```diff
@@ -1,5 +1,9 @@
 # frozen_string_literal: true
 
 class Destination < ActiveRecord::Base
-  validates :url, presence: true
+  enum service_type: {growth_spaces: "growth_spaces", adobe_campaigns: "adobe_campaigns"}
+
+  validates :url, :service_type, presence: true
+  validates :route_id, presence: true, if: :growth_spaces?
+  validates :service_name, presence: true, if: :adobe_campaigns?
 end
```

Comment:
I wonder should we add:
`validates :access_key_id, :access_key_secret presence: true`
or 
`validates :access_key_id, :access_key_secret presence: true, if :growth_spaces? or :adobe_campaigns?` (syntax??)

Reply:
@hlbraddock I also think we should add the validation. 
The second option is not really necessary, since those are the only two possible options at the time.